test/listener/server-profile-templates.js

```diff
@@ -0,0 +1,225 @@
+    'bound UnDeployProfiles'.should.equal(constructorArgs[5].args[2].name);
+    assert.isFalse(constructorArgs[5].args[2].hasOwnProperty('prototype'));
+
+    rgx6.should.deep.equal(constructorArgs[6].args[0]);
+    assert(typeof constructorArgs[6].args[2] === 'function');
+    'bound UnDeployProfiles'.should.equal(constructorArgs[6].args[2].name);
+    assert.isFalse(constructorArgs[6].args[2].hasOwnProperty('prototype'));
+
+    rgx7.should.deep.equal(constructorArgs[7].args[0]);
+    assert(typeof constructorArgs[7].args[2] === 'function');
+    'bound FixCompliance'.should.equal(constructorArgs[7].args[2].name);
+    assert.isFalse(constructorArgs[7].args[2].hasOwnProperty('prototype'));
+
+    assert(constructorArgs.length === 8);
+
+    done();
+  });
+
+  it('test MakeProfileCompliant for unassigned profile', (done) => {
+
+    const profile = new ServerProfile(profileResponse, brain);
+
+    const serverProfileTemplateListener = new ServerProfileTemplateListener(robot, client, transform, sh, serverProfilesListener, brain);
+
+    let msg = {
+        host: {},
+        serverId: 1
+    };
+
+    let spy1 = sinon.spy(serverProfileTemplateListener, "PowerOffHardware");
+    let spy2 = sinon.spy(serverProfilesListener, "MakeServerProfileCompliant");
+
+    let stub1 = sinon.stub(client.ServerProfiles, 'updateServerProfileCompliance').returns(
+        new PromiseFeedback((feedback) => {
+            return Bluebird.resolve(profileResponse);
+        })
+    );
+
+    serverProfileTemplateListener.MakeProfileCompliant(profile, msg);
+
+    assert(serverProfileTemplateListener.PowerOffHardware.callCount === 0);
+    assert(serverProfilesListener.MakeServerProfileCompliant.callCount === 1);
+    //console.log(serverProfilesListener.MakeServerProfileCompliant);
+
+    spy1.restore();
+    spy2.restore();
+    stub1.restore();
+
+    done();
+  });
+
+  it('test MakeProfileCompliant for assigned profile', (done) => {
+
+    profileResponse.serverHardwareUri = "/rest/server-profiles/1a94be5c";
+    const profile = new ServerProfile(profileResponse, brain);
+
+    const serverProfileTemplateListener = new ServerProfileTemplateListener(robot, client, transform, sh, serverProfilesListener, brain);
+    let sp1 = sinon.spy(serverProfileTemplateListener, 'PowerOffHardware');
+
+    let msg = {
+        host: {},
+        serverId: 1,
+        send: function() {}
+    };
+
+    let st1 = sinon.stub(client.ServerHardware, 'setPowerState').returns(
+        new PromiseFeedback((feedback) => {
+            return Bluebird.resolve(profileResponse);
+        })
+    );
+
+    let st2 = sinon.stub(serverProfilesListener, 'MakeServerProfileCompliant').returns(
+        new PromiseFeedback((feedback) => {
+            return Bluebird.resolve(profileResponse);
+        })
+    );
+
+    serverProfileTemplateListener.MakeProfileCompliant(profile, msg);
+
+    assert(sp1.calledOnce);
+
+    sp1.restore();
+    
```

Comment:
probably want to call restore on st1 and st2

Reply:
I tried to.  It threw an exception